File: Irrgarten_ruby/fuzzy_player.rb
# frozen_string_literal: true

module Irrgarten

  class FuzzyPlayer < Player

    def initialize(p)#Constructor de copia

      super(p.numero, p.intelligence, p.strength)
    end

    def move(direction, valid_moves)

      Dice.next_step(direction,valid_moves,@intelligence) #super.intelligence
    end

    def attack

      Dice.intensity(@strength+sum_weapons) #super.strength y super.sum_weapons
    end

    def defense_energy

      Dice.intensity(@intelligence+sum_shields) #super.intelligence y super.sum_shields
    end

    def to_s

      "Fuzzy #{super.to_s}"
    end
  end
end

File: TestP1/src/irrgarten/FuzzyPlayer.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

import java.util.ArrayList;

/**
 *
 * @author zapi24
 */
public class FuzzyPlayer extends Player{
    

    public FuzzyPlayer(Player p) {
        super(p.getNumero(), p.getIntelligence(), p.getStrength());
    }
    
    @Override //Se mueve de forma distinta que un player 
    public Directions move(Directions direction, ArrayList<Directions> validMoves){
        
        
        return Dice.nextStep(direction, validMoves, super.getIntelligence());
    }
    
    @Override
    public float attack(){
        
        return Dice.intensity(super.getStrength())+super.sumWeapons();
    }
    
    protected float defenseEnergy(){
        
        return Dice.intensity(super.getIntelligence())+super.sumShields();
    }
    
    @Override
    public String toString(){
        
        return "Fuzzy " + super.toString();
    }
    
    
    
}
